src/nbl/asset/utils/ISPIRVEntryPointTrimmer.cpp

```diff
@@ -0,0 +1,260 @@
+#include "nbl/asset/utils/ISPIRVEntryPointTrimmer.h"
+#include "nbl/asset/utils/ISPIRVOptimizer.h"
+#include "nbl_spirv_cross/spirv.hpp"
+
+#include "nbl/core/declarations.h"
+#include "nbl/system/ILogger.h"
+#include "spirv-tools/libspirv.hpp"
+
+using namespace nbl::asset;
+
+static constexpr spv_target_env SPIRV_VERSION = spv_target_env::SPV_ENV_UNIVERSAL_1_6;
+
+ISPIRVEntryPointTrimmer::ISPIRVEntryPointTrimmer()
+{
+    // Multiple dead branch and dead function elimination because the first entry point removal might result to dead branch. Then the dead branch might result to dead function. Then, the dead function might result to dead branch and so on.
+    constexpr auto optimizationPasses = std::array{
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
+        ISPIRVOptimizer::EOP_DEAD_BRANCH_ELIM,
+        ISPIRVOptimizer::EOP_ELIM_DEAD_FUNCTIONS,
```

Comment:
wouldn't that mean you have to run dead branch and dead function elimination on a loop until SPIR-V no longer changes?